### PR TITLE
 Add search for scaffolds page

### PR DESCRIPTION
**Pull request by evenstensberg**
This is the initial PR of the upcoming documentation efforts for webpack and webpack-cli. It will allow people to search for scaffolds as well as browsering for exsisting scaffolds. This will be used for webpack-ui to generate configurations. What is missing is the css for the search page as well as design for the result cards and implementation of these cards 👍

### CODE REVIEW

File: src/components/Scaffolding/Scaffolding.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    background-color: getColor(elephant);
+}
+
+.ais-Hits-item {
+    flex: 0 0 100%;
+}
+.product-wrapper {
+    border-top: 1px solid getColor(denim);
+    border-bottom: 1px solid getColor(denim);
+    margin: 5px;
```
Comment (rishabh3112) [inline src/components/Scaffolding/Scaffolding.scss]: Add margin left and right of 10px

Would make cards differentiable as compared to outside text.

**Comment by montogeek** [comment]
@EugeneHlushko We need to merge `rebuild` fast, then update `next`.

**Comment by evenstensberg** [comment]
Need help on design here. Take the PR and edit CSS/SCSS and you are ready. Link to the repo on click on the card.

**Comment by jahzielv** [comment]
Hi @evenstensberg, I'd be interested in helping here! Can you elaborate a little more on what needs to be done?

**Comment by evenstensberg** [comment]
@jahzielv Nice, thanks! If you run the project locally, you can see the design is bad. You can implement a same design such as https://yeoman.io/generators/.

**Comment by anshumanv** [comment]
> Need help on design here. Take the PR and edit CSS/SCSS and you are ready. Link to the repo on click on the card.

On it. 👍

**Comment by evenstensberg** [comment]
@webpack/documentation-team PTAL

File: src/components/Scaffolding/HitBox.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Highlight } from 'react-instantsearch-dom';
+
+class HitBox extends React.Component {
+    render() {
+        const { hit } = this.props;
+        return (
+        <article>
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/HitBox.jsx]: Please align this consistently with other jsx files in the project

File: src/components/Scaffolding/Scaffolding.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.jsx]:
```suggestion
```

File: src/components/Scaffolding/Scaffolding.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import React from 'react';
+
+import './Scaffolding.scss';
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.jsx]: Can we move local relative path imports to after node_modules imports?

File: src/components/Scaffolding/Scaffolding.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  function htmlDecode(input){
+    //eslint-disable-next-line
+    return input.replace(/[^a-z0-9\s]/gi, '').replace(/(gt)/gi, '')
+  }
+  const Hit = ({ hit }) => {
+  
+    return (
+      <article className="hit-card">
+        <a target="_blank" rel="noopener noreferrer"
+          href={hit.repository && hit.repository.url ? hit.repository.url: '/'}>
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.jsx]:
```suggestion
          href={hit.repository && hit.repository.url ? hit.repository.url : '/'}>
```

File: src/components/Scaffolding/Scaffolding.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+          <div className="product-wrapper">
+              <div className="product-summary">
+                <div className="product-title">
+                <h3>{hit.name}</h3>
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.jsx]:
```suggestion
                  <h3>{hit.name}</h3> 
```

File: src/components/Scaffolding/Scaffolding.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+              <div className="product-summary">
+                <div className="product-title">
+                <h3>{hit.name}</h3> 
+                </div>
+                <div className="product-avatar">
+                  <img src={hit.owner.avatar} height="40" width="40"/>
+                </div>
+              </div>
+              <div className="product-name">
+              <h3>{hit.owner.name}</h3>
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.jsx]:
```suggestion
                <h3>{hit.owner.name}</h3>
```

File: src/components/Scaffolding/Scaffolding.scss [hunk 8]
```diff
@@ -0,0 +1,88 @@
+@import 'vars';
+@import 'mixins';
+@import 'functions';
+
+/* fiord, elephant, alto, malibu, concrete, mine-shaft */
+
+
+.ais-Hits-list  {
+    list-style: none;
+    display:flex;
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.scss]:
```suggestion
    display: flex;
```

File: src/components/Scaffolding/Scaffolding.scss [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    flex-direction:column;
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.scss]:
```suggestion
    flex-direction: column;
```

File: src/components/Scaffolding/Scaffolding.scss [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    justify-content: flex-end;
+}
+.ais-SearchBox-input {
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    appearance: none;
+    padding: .3rem 1.7rem;
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.scss]: Can we use `em` as the rest of the modules?

File: src/components/Scaffolding/Scaffolding.scss [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+
+.ais-Hits-list  {
+    list-style: none;
+    display:flex;
+    flex-direction:column;
+    justify-content: flex-end;
+}
+.ais-SearchBox-input {
+    -webkit-appearance: none;
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.scss]: Remove browser prefixes from scss

File: src/components/Scaffolding/Scaffolding.scss [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    display: flex;
+    float: right;
+}
+
+.product-title {
+    padding: 5px 15px 5px 15px;
+}
+
+.product-avatar {
+    & img {
```
Comment (EugeneHlushko) [inline src/components/Scaffolding/Scaffolding.scss]: What did you mean wtih `& img`, wont this compile to `img .product-avatar {border-radius:...}` ?

File: src/components/Page/Page.jsx [hunk 13]
```diff
@@ -79,7 +80,7 @@ const Page = ({ page, section }) => {
             </ul>
           </div>
         )}
-
+        {title === "Discover scaffolds" ? <Scaffolding /> : null}
```
Comment (EugeneHlushko) [inline src/components/Page/Page.jsx]:
```suggestion
        {title === 'Discover scaffolds' ? <Scaffolding /> : null}
```

**Comment by evenstensberg** [comment]
Ptal

**Comment by misterdev** [comment]
@EugeneHlushko & @skipjack, do you mind looking into this? :)

File: src/components/Scaffolding/Scaffolding.jsx [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    </article>
+  );
+};
+
+export default class Scaffolding extends React.Component {
+  render() {
+     return (
+      <div>
+        <InstantSearch searchClient={searchClient} indexName="npm-search">
+          <Configure filters="computedKeywords:webpack-scaffold" hitsPerPage={16} />
```
Comment (Haroenv) [inline src/components/Scaffolding/Scaffolding.jsx]:
```suggestion
          <Configure
            filters="computedKeywords:webpack-scaffold"
            hitsPerPage={16}
            analyticsTags={['webpack-scaffold']}
          />
```

File: src/components/Scaffolding/Scaffolding.jsx (hunk 4 above)
Comment (evenstensberg) [reply src/components/Scaffolding/Scaffolding.jsx]: Fixed

File: src/components/Scaffolding/Scaffolding.scss (hunk 10 above)
Comment (evenstensberg) [reply src/components/Scaffolding/Scaffolding.scss]: yes, fixed

**Comment by evenstensberg** [comment]
Ptal

**Comment by EugeneHlushko** [comment]
Can you create new PR so we get netlify in here for previews please

**Comment by montogeek** [comment]
Open a new PR just for the sake of a preview is bad :/
Conversation and feedback get a bit lost, I know it can be linked, but still :/

**Comment by montogeek** [comment]
@EugeneHlushko It is not possible to trigger it from Netlify?

**Comment by EugeneHlushko** [comment]
I've checked and looks like i cannot do it, only allows triggering build for master on free tier UI

**Comment by EugeneHlushko** [comment]
the PR is quite old, i think this time we can lose convo...

**Pull request closed without merging**